src/cascadia/TerminalCore/TerminalDispatch.cpp

```diff
@@ -786,3 +789,84 @@ bool TerminalDispatch::HardReset()
 
     return true;
 }
+
+// Routine Description:
+// - DECSC - Saves the current "cursor state" into a memory buffer. This
+//   includes the cursor position, origin mode, graphic rendition, and
+//   active character set.
+// Arguments:
+// - <none>
+// Return Value:
+// - True.
+bool TerminalDispatch::CursorSaveState()
+{
+    // TODO GH#3849: When de-duplicating this, the AdaptDispatch version of this
+    // is more elaborate.
+    const auto attributes = _terminalApi.GetTextAttributes();
+    COORD coordCursor = _terminalApi.GetCursorPosition();
+    // The cursor is given to us by the API as relative to current viewport top.
+
+    // VT is also 1 based, not 0 based, so correct by 1.
+    auto& savedCursorState = _savedCursorState.at(_usingAltBuffer);
+    savedCursorState.Column = coordCursor.X + 1;
+    savedCursorState.Row = coordCursor.Y + 1;

```

Comment:
it seems like DECSC DECRC DECSC DECRC will move the cursor down right down right a few times because of the adjustment here?

I guess what i'm saying is.. i don't understand how VT being 1-based matters here, as they did not give us a coordinate in the request; the numbers we are using are all internal to us

Reply:
it might, but DECRC & DECSC are definitely not hooked all the way up yet so that's not an issue yet?